lib/pry/cli.rb

```diff
@@ -59,11 +59,19 @@ def parse_options(args = ARGV)
           raise NoOptionsError, "No command line options defined! Use Pry::CLI.add_options to add command line options."
         end
 
-        self.input_args = args
+        index = args.index { |cli_arg| %w[- --].include?(cli_arg) }
+        if index
+          @pass_argv = true
+          slop_args = args[0...index]
+          self.input_args = args.replace(args[index + 1..-1])
+        else
+          @pass_argv = false
```

Comment:
Is this variable truly needed? On line 112 we can check for `index`. I would actually rename `index` to `pass_argv` and use it.

Reply:
I assume you meant rename `index` to pass `@pass_argv` ..? 
Line 112 is in another method, that's why an instance variable is being used. I agree we can drop `index` for a single ivar instead.